### PR TITLE
feat: Dynamic status tags for menu items

### PR DESCRIPTION
**Pull request by memoyil**
Feature for Menu item tags (currently for ifo only) to be placed automatically based on component status

### CODE REVIEW

File: src/views/Home/components/Banners/hooks/useIsRenderIFOBanner.tsx [hunk 1]
```diff
@@ -1,17 +1,12 @@
...
 import { useActiveIfoWithBlocks } from 'hooks/useActiveIfoWithBlocks'
-import { getStatus } from 'views/Ifos/hooks/helpers'
 
 const useIsRenderIfoBanner = () => {
   const currentBlock = useCurrentBlock()
 
   const activeIfoWithBlocks = useActiveIfoWithBlocks()
 
-  const isIfoAlive = !!(currentBlock && activeIfoWithBlocks && activeIfoWithBlocks.endBlock > currentBlock)
-  const status = isIfoAlive
```
Comment (memoyil) [inline src/views/Home/components/Banners/hooks/useIsRenderIFOBanner.tsx:11]: isIfoAlive is enough to determine rendering

File: src/components/Menu/index.tsx [hunk 2]
```diff
@@ -42,7 +45,7 @@ const Menu = (props) => {
       langs={languageList}
       setLang={setLanguage}
       cakePriceUsd={cakePriceUsd.toNumber()}
-      links={config(t)}
+      links={config(t, menuItemsStatus)}
```
Comment (chefilip) [inline src/components/Menu/index.tsx:48]: I think we could turn config into a hook to avoid re-rendering

**Review by chefilip — COMMENTED** [review]
Fantastic. Thanks for doing this ❤️

File: src/hooks/useMenuItemsStatus.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  const ifoStatus =
+    currentBlock && activeIfo && activeIfo.endBlock > currentBlock
+      ? getStatus(currentBlock, activeIfo.startBlock, activeIfo.endBlock)
+      : null
+
+  return ifoStatus
+    ? {
+        '/ifo': ifoStatus === 'coming_soon' ? 'soon' : ifoStatus,
+      }
+    : null
```
Comment (chef-jojo) [inline src/hooks/useMenuItemsStatus.ts:18]: let's use memo

File: src/components/Menu/index.tsx [hunk 4]
```diff
@@ -24,6 +25,8 @@ const Menu = (props) => {
   const activeMenuItem = getActiveMenuItem({ menuConfig: config(t), pathname })
```
Comment (chef-jojo) [inline src/components/Menu/index.tsx:25]: here can lift it to useMenuConfig or something

**Comment by chef-jojo** [comment]
@memoyil, thanks for the PR.
I have an idea to inject hook inside the menu items. Having a currentBlock hook on top of menu looks a bit scary, but I think it's fine for now.

File: src/hooks/useMenuItemsStatus.ts (hunk 3 above)
Comment (memoyil) [reply src/hooks/useMenuItemsStatus.ts]: Done

File: src/components/Menu/index.tsx (hunk 2 above)
Comment (memoyil) [reply src/components/Menu/index.tsx]: Done

File: src/components/Menu/index.tsx (hunk 4 above)
Comment (memoyil) [reply src/components/Menu/index.tsx]: Done

**Comment by memoyil** [comment]
> @memoyil, thanks for the PR. I have an idea to inject hook inside the menu items. Having a currentBlock hook on top of menu looks a bit scary, but I think it's fine for now.

I think so too, we have cakePriceUsd in that menu that is updated with fast refresh same as block number and there is no/negligible performance impact

**Pull request merged**